www/includes/easyparliament/templates/html/mp/divisions.php

```diff
@@ -50,17 +52,29 @@
                         # does all divisions, but generally this is an array of one
                         # (the current policy)
                         foreach ($policydivisions as $policy) { ?>
-
+                        
                             <?php
+                                $current_policy_agreements = $policyagreements[$policy['policy_id']] ?? [];
```

Comment:
I realise these aren't perfect, but generally I'd try and put the "?? []" bit of this in the business code that fetched the data, so the template can worry less about whether things are defined or not.

Reply:
So retracing my steps, I think I do that in this bit here when we know the policy_id: 

https://github.com/mysociety/theyworkforyou/blob/8d5c0e389604dfed0d8d63c8c044d1b9e64e3101/www/docs/mp/index.php#L443-L455

BUT - there'd then be a problem for the 'all policies' page (which I think is mostly unused but theoretically there) -because not every policy has agreements and so most of that array is unset and so empty. 

So rather than :

```php
$data['policyagreements'] = $policiesList->all_policy_agreements;
```

It would be the following to make sure that policy_ids without agreements get empty lists:

```php
$policy_to_agreements = [];
foreach ($policies->getPolicies() as $policy_id => $policy_text) {
    $policy_to_agreements[$policy_id] = $policiesList->all_policy_agreements[$policy_id] ?? [];
}

$data['policyagreements'] = $policy_to_agreements;
```

And actually that could be put all the way back where `all_policy_agreements` is first defined, and lose a lot of these tests - because they'll all have empty lists to start with. 

I'll test that out later - but just checking the basic idea.

Reply:
OIC, yes, as you're just passing in the whole thing then. It's fine in the template :)